TTHAnalysis/python/tools/nanoAOD/fastCombinedObjectRecleaner.py

```diff
@@ -91,13 +91,16 @@ def initReaders(self,tree):
             if coll==self.tauc: _vars.extend(self.vars_taus+self.vars_taus_int+self.vars_taus_uchar)
             if coll==self.jc: _vars.extend(self.vars_jets+self.vars_jets_int+self.vars_jets_nooutput)
             for B in _vars:
-                setattr(self,"%s_%s"%(coll,B), tree.arrayReader("%s_%s"%(coll,B)))
+                if type(B) == tuple:
+                    setattr(self,"%s_%s"%(coll,B[1]), tree.arrayReader("%s_%s"%(coll,B[1])))
+                else:
+                    setattr(self,"%s_%s"%(coll,B), tree.arrayReader("%s_%s"%(coll,B)))
         return True
 
     def initWorkers(self):
         self._worker.setLeptons(self.nLepGood, self.LepGood_pt, self.LepGood_eta, self.LepGood_phi, self.LepGood_jetIdx)
         self._worker.setTaus(getattr(self,'n%s'%self.tauc),getattr(self,'%s_pt'%self.tauc),getattr(self,'%s_eta'%self.tauc),getattr(self,'%s_phi'%self.tauc), getattr(self,'%s_jetIdx'%self.tauc))
-        self._worker.setJets(getattr(self,'n%s'%self.jc),getattr(self,'%s_pt'%self.jc),getattr(self,'%s_eta'%self.jc),getattr(self,'%s_phi'%self.jc),
+        self._worker.setJets(getattr(self,'n%s'%self.jc),getattr(self,'%s_pt_nom'%self.jc),getattr(self,'%s_eta'%self.jc),getattr(self,'%s_phi'%self.jc),
```

Comment:
and here `_pt`

otherwise, just use `('%s_pt_nom' if self.isMC else '%s_pt')` here, and then at the beginnig you can get rid of the tuple and simply put `('pt_nom' if self.isMC else 'pt')` there, or define at the beginning `self._jetPtName = ('pt_nom' if self.isMC else 'pt')` and use that in both places